src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs

```diff
@@ -613,19 +609,21 @@ public DisplayVersion SelectedVersion
                         {
                             PackageMetadata = detailedPackageMetadata;
                         }
-
-                        NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+                        else
```

Comment:
Is this fixing another bug with the version dropdown?

Reply:
No, not fixing a bug persay. If you look at SelectedVersionChangedAsync it ends setting the PackageMetadata property again. 

I didn't think it was necessary to try to retrieve the metadata information and update the _metadataDict cache after we've already used it.